internal/orchestration/controller.go

```diff
@@ -342,14 +343,31 @@ func (e *Experiment) stopReplicas() error {
 		for id, hash := range res.GetHashes() {
 			hashes[id] = hash
 		}
+		for id, cnt := range res.GetCounts() {
+			counts[id] = cnt
+		}
+	}
+	for id1, cnt1 := range counts {
+		for id2, cnt2 := range counts {
+			if id2 > id1 && cnt2 == cnt1 {
+				if !bytes.Equal(hashes[id1], hashes[id2]) {
+					return fmt.Errorf("hash mismatch for same count")
+				}
+			}
+		}
 	}
-	var cmp []byte
-	for _, hash := range hashes {
-		if cmp == nil {
-			cmp = hash
+	var count uint32
+	first := true
+	for _, cnt := range counts {
+		if cnt == 0 {
+			return fmt.Errorf("no request executed")
+		}
+		if first {
```

Comment:
Line 359 could be replaced with the following:
```go
var count uint32
if len(counts) > 0 {
   count = counts[0]
}
```
Avoiding the `first` bool and Lines 365-368.